hypertrace-trace-enricher/hypertrace-trace-enricher-impl/src/test/java/org/hypertrace/traceenricher/enrichment/enrichers/ResourceAttributeEnricherTest.java

```diff
@@ -0,0 +1,243 @@
+
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+public class ResourceAttributeEnricherTest extends AbstractAttributeEnricherTest {
+
+  private final ResourceAttributeEnricher resourceAttributeEnricher =
+      new ResourceAttributeEnricher();
+
+  @BeforeAll
+  public void setup() {
+    String configFilePath =
+        Thread.currentThread().getContextClassLoader().getResource("enricher.conf").getPath();
+    if (configFilePath == null) {
+      throw new RuntimeException("Cannot find enricher config file enricher.conf in the classpath");
+    }
+
+    Config fileConfig = ConfigFactory.parseFile(new File(configFilePath));
+    Config configs = ConfigFactory.load(fileConfig);
+    if (!configs.hasPath("enricher.ResourceAttributeEnricher")) {
+      throw new RuntimeException(
+          "Cannot find enricher config for ResourceAttributeEnricher in " + configs);
+    }
+    resourceAttributeEnricher.init(
+        configs.getConfig("enricher.ResourceAttributeEnricher"), mock(ClientRegistry.class));
+  }
+
+  @Test
+  public void noAttributes() {
+    Event event = mock(Event.class);
+    when(event.getAttributes()).thenReturn(null);
+    resourceAttributeEnricher.enrichEvent(null, event);
+
+    when(event.getAttributes()).thenReturn(Attributes.newBuilder().build());
+    when(event.getAttributes().getAttributeMap()).thenReturn(null);
+    resourceAttributeEnricher.enrichEvent(null, event);
+  }
+
+  @Test
+  public void noResourceInTrace() {
+    // This trace has no resource attributes.
+    StructuredTrace trace = getBigTrace();
+    for (Event event : trace.getEventList()) {
+      int attributeMapSize = 0;
+      if (event.getAttributes() != null && event.getAttributes().getAttributeMap() != null) {
+        attributeMapSize = event.getAttributes().getAttributeMap().size();
+      }
+      resourceAttributeEnricher.enrichEvent(trace, event);
+      if (event.getAttributes() != null && event.getAttributes().getAttributeMap() != null) {
+        assertEquals(attributeMapSize, event.getAttributes().getAttributeMap().size());
+      }
+    }
+  }
+
+  @Test
+  public void traceWithResource() {
+    StructuredTrace structuredTrace = mock(StructuredTrace.class);
+    List<Resource> resourceList = new ArrayList<>();
+
+    resourceList.add(getResource1());
+    resourceList.add(getResource2());
+    resourceList.add(getResource3());
+    resourceList.add(getResource4());
+    when(structuredTrace.getResourceList()).thenReturn(resourceList);
+
+    Attributes attributes = Attributes.newBuilder().setAttributeMap(new HashMap<>()).build();
+    Event event =
+        Event.newBuilder()
+            .setAttributes(attributes)
+            .setEventId(createByteBuffer("event1"))
+            .setCustomerId(TENANT_ID)
+            .build();
+    event.setResourceIndex(0);
+    resourceAttributeEnricher.enrichEvent(structuredTrace, event);
+    assertEquals(
+        resourceAttributeEnricher.getResourceAttributesToAdd().size() - 2,
```

Comment:
Why is this method in the public API? Seems like it has been added just to facilitate this test.